client/state/sites/test/selectors.js

```diff
@@ -1184,4 +1228,778 @@ describe( 'selectors', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: false
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( null );
+		} );
+
+		it( 'it should return `true` if jetpack version is strictly less than 3.4', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						jetpack_version: '3.3'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( true );
+		} );
+
+		it( 'it should return `true` if the modules has not yet been fetched', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						active_modules: [],
+						jetpack_version: '3.4'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( true );
+		} );
+
+		it( 'it should return `true` if jetpack version is greater or equal to 3.4 and the manage module is active', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						active_modules: [ 'manage' ],
+						jetpack_version: '3.4'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( true );
+		} );
+
+		it( 'it should return `false` if jetpack version is greater or equal to 3.4 and the manage module is not active', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						active_modules: [ 'sso' ],
+						jetpack_version: '3.4'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( false );
+		} );
+	} );
+
+	describe( '#canJetpackSiteUpdateFiles()', () => {
+		let configStub;
+
+		useSandbox( ( sandbox ) => {
+			configStub = sandbox.stub().withArgs( 'jetpack_min_version' ).returns( '3.3' );
+		} );
+
+		it( 'should return `null` for a non-existing site', () => {
+			const canUpdateFiles = canJetpackSiteUpdateFiles( stateWithNoItems, nonExistingSiteId );
+			expect( canUpdateFiles ).to.equal( null );
+		} );
+
+		it( 'it should return `false` for a non jetpack site', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: false,
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( null );
+		} );
+
+		it( 'it should return `false` if jetpack version is smaller than minimum version', () => {
+			const jetpackMinVersion = configStub( 'jetpack_min_version' );
+			const smallerVersion = changeVersion( jetpackMinVersion, -1 );
```

Comment:
I don't, really. We're explicitly setting `jetpack_min_version` in the config stub anyway, so why not make the "lesser version" explicit, too (instead of adding a somewhat complex function to decrement that version)? I'd much rather remove that additional utility function, TBH.